antarest/study/storage/rawstudy/model/filesystem/lazy_node.py

```diff
@@ -83,35 +85,54 @@ def _get(
 
     def get(
         self,
-        url: Optional[List[str]] = None,
+        url: t.Optional[t.List[str]] = None,
         depth: int = -1,
         expanded: bool = False,
         formatted: bool = True,
-    ) -> Union[str, G]:
+    ) -> t.Union[str, G]:
         output = self._get(url, depth, expanded, formatted, get_node=False)
         assert not isinstance(output, INode)
         return output
 
     def get_node(
         self,
-        url: Optional[List[str]] = None,
+        url: t.Optional[t.List[str]] = None,
     ) -> INode[G, S, V]:
         output = self._get(url, get_node=True)
         assert isinstance(output, INode)
         return output
 
-    def delete(self, url: Optional[List[str]] = None) -> None:
+    def delete(self, url: t.Optional[t.List[str]] = None) -> None:
         self._assert_url_end(url)
         if self.get_link_path().exists():
             self.get_link_path().unlink()
         elif self.config.path.exists():
             self.config.path.unlink()
 
+    def infer_path(self) -> Path:
+        if self.get_link_path().exists():
+            return self.get_link_path()
+        elif self.config.path.exists():
+            return self.config.path
+        else:
+            raise ChildNotFoundError(
+                f"Neither link file {self.get_link_path} nor matrix file {self.config.path} exists"
+            )
+
+    def infer_target_path(self, is_link: bool) -> Path:
+        if is_link:
+            return self.get_link_path()
+        else:
+            return self.config.path
+
+    def infer_is_link_path(self) -> bool:
```

Comment:
remove this method, it's only used once and it's just one line. Also I think you should remove the `infer_targer_path` method and adapt your code inside `move_file_to` line 160